cli/src/merge_tools/diff_working_copies.rs

```diff
@@ -177,50 +176,55 @@ pub(crate) fn check_out_trees(
         left_tree_state,
         right_tree_state,
         output_tree_state,
-        instructions_path_to_cleanup: None,
     })
 }
 
-/// Checks out the trees, populates JJ_INSTRUCTIONS, and makes appropriate sides
-/// readonly.
-pub fn check_out_trees_for_diffedit(
-    store: &Arc<Store>,
-    left_tree: &MergedTree,
-    right_tree: &MergedTree,
-    matcher: &dyn Matcher,
-    output_is: Option<DiffSide>,
-    instructions: Option<&str>,
-) -> Result<DiffWorkingCopies, DiffEditError> {
-    let mut diff_wc = check_out_trees(store, left_tree, right_tree, matcher, output_is)?;
-    let got_output_field = output_is.is_some();
+pub(crate) struct DiffEditWorkingCopies {
+    pub working_copies: DiffWorkingCopies,
+    instructions_path_to_cleanup: Option<PathBuf>,
+}
 
-    set_readonly_recursively(diff_wc.left_working_copy_path())
-        .map_err(ExternalToolError::SetUpDir)?;
-    if got_output_field {
-        set_readonly_recursively(diff_wc.right_working_copy_path())
+impl DiffEditWorkingCopies {
+    /// Checks out the trees, populates JJ_INSTRUCTIONS, and makes appropriate
+    /// sides readonly.
+    pub fn check_out_trees_for_diffedit(
```

Comment:
nit: rename to `check_out()` or `check_out_trees()` as it is now scoped?

Reply:
Done.